lib/manageiq/appliance_console/message_configuration_server.rb

```diff
@@ -0,0 +1,224 @@
+
+        return if file_found?(LOGS_DIR)
+
+        FileUtils.mkdir_p(LOGS_DIR)
+        FileUtils.chmod(0o755, LOGS_DIR)
+        FileUtils.chown("kafka", "kafka", LOGS_DIR)
+      end
+
+      def configure_firewall
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        AwesomeSpawn.run!("firewall-cmd --add-port=9093/tcp --permanent") # secure
+        AwesomeSpawn.run!("firewall-cmd --reload")
+      end
+
+      def configure_keystore
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        return if file_found?(keystore_dir_path)
+
+        FileUtils.mkdir_p(keystore_dir_path)
+        FileUtils.chmod(0o755, keystore_dir_path)
+
+        # Generte a Java keystore and key pair, creating keystore.jks
+        AwesomeSpawn.run!("keytool", :params => {"-keystore" => keystore_path, "-alias" => "localhost", "-validity" => 10_000, "-genkey" => nil, "-keyalg" => "RSA", "-storepass" => password, "-keypass" => password, "-dname" => "cn=#{server_hostname}"})
+
+        # Use openssl to create a new CA cert, creating ca-cert and ca-key
+        AwesomeSpawn.run!("openssl", :env => {"PASSWORD" => password}, :params => ["req", "-new", "-x509", {"-keyout" => ca_key_path, "-out" => ca_cert_path, "-days" => 10_000, "-passout" => "env:PASSWORD", "-subj" => '/CN=something'}])
+
+        # Import the CA cert into the trust store, creating truststore.jks
+        AwesomeSpawn.run!("keytool", :params => {"-keystore" => truststore_path, "-alias" => "CARoot", "-import" => nil, "-file" => ca_cert_path, "-storepass" => password, "-noprompt" => nil})
+
+        # Generate a certificate signing request (CSR) for an existing Java keystore, creating cert-file
+        AwesomeSpawn.run!("keytool", :params => {"-keystore" => keystore_path, "-alias" => "localhost", "-certreq" => nil, "-file" => cert_file_path, "-storepass" => password})
+
+        # Use openssl to sign the certificate with the "CA" certificate, creating ca-cert.srl and cert-signed
+        AwesomeSpawn.run!("openssl", :env => {"PASSWORD" => password}, :params => ["x509", "-req", {"-CA" => ca_cert_path, "-CAkey" => ca_key_path, "-in" => cert_file_path, "-out" => cert_signed_path, "-days" => 10_000, "-CAcreateserial" => nil, "-passin" => "env:PASSWORD"}])
+
+        # Import a root or intermediate CA certificate to an existing Java keystore, updating keystore.jks
+        AwesomeSpawn.run!("keytool", :params => {"-keystore" => keystore_path, "-alias" => "CARoot", "-import" => nil, "-file" => ca_cert_path, "-storepass" => password, "-noprompt" => nil})
+
+        # Import a signed primary certificate to an existing Java keystore, updating keystore.jks
+        AwesomeSpawn.run!("keytool", :params => {"-keystore" => keystore_path, "-alias" => "localhost", "-import" => nil, "-file" => cert_signed_path, "-storepass" => password, "-noprompt" => nil})
+      end
+
+      def create_server_properties
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        content = <<~SERVER_PROPERTIES
```

Comment:
@bdunne Yes there are roughly 20 name=value pairs already in the server.properties file. I think doing it this way could be safer than hardcoding what is delivered in the file because future versions of kafka could be delivered with different content. Then we would have to update the content here too.
I propose we leave this as is.